src/NodeApi/JSValue.cs

```diff
@@ -21,38 +20,56 @@ namespace Microsoft.JavaScript.NodeApi;
 
     internal JSRuntime Runtime => Scope.Runtime;
 
+    /// <summary>
+    /// Creates an empty instance of <see cref="JSValue" />, which implicitly converts to
+    /// <see cref="JSValue.Undefined" /> when used in any scope.
+    /// </summary>
     public JSValue() { }
 
-    public JSValue(napi_value handle) : this(handle, JSValueScope.Current)
-    {
-    }
-
+    /// <summary>
+    /// Creates a new instance of <see cref="JSValue" /> from a handle in the specified scope.
+    /// </summary>
+    /// <exception cref="ArgumentNullException">Thrown when the scope is null (unless the handle
+    /// is also null).</exception>
+    /// <remarks>
+    /// WARNING: A JS value handle is a pointer to a location in memory, so an invalid handle here
+    /// may cause an attempt to access an invalid memory location.
+    /// </remarks>
     public JSValue(napi_value handle, JSValueScope? scope)
     {
         if (!handle.IsNull && scope is null) throw new ArgumentNullException(nameof(scope));
         _handle = handle;
         _scope = scope;
     }
 
-    public napi_value? Handle
```

Comment:
We are removing a check that current Handle is valid. What is an alternative to that check?

Reply:
I added the checks back to here, so now getting either the `napi_value` or `napi_env` is checked.